tests/persistence/test_persistence_leverage.py

```diff
@@ -0,0 +1,638 @@
+        fee_open=fee.return_value,
+        fee_close=fee.return_value,
+        exchange='kraken',
+        leverage=3.0,
+        interest_rate=0.0005,
+        interest_mode=InterestMode.HOURSPER4
+    )
+
+    assert float(trade.calculate_interest()) == 7.5414886436454e-06
+    trade.open_date = datetime.utcnow() - timedelta(hours=4, minutes=55)
+    assert float(round(trade.calculate_interest(interest_rate=0.00025), 11)
+                 ) == round(5.65611648273405e-06, 11)
+
+    trade = Trade(
+        pair='ETH/BTC',
+        stake_amount=0.0037707443218227,
+        amount=459.95905365,
+        open_rate=0.00001099,
+        open_date=datetime.utcnow() - timedelta(hours=4, minutes=55),
+        fee_open=fee.return_value,
+        fee_close=fee.return_value,
+        exchange='kraken',
+        leverage=5.0,
+        interest_rate=0.0005,
+        interest_mode=InterestMode.HOURSPER4
+    )
+
+    assert float(round(trade.calculate_interest(), 11)
+                 ) == round(2.26244659309362e-05, 11)
+    trade.open_date = datetime.utcnow() - timedelta(hours=0, minutes=10)
+    trade.interest_rate = 0.00025
+    assert float(trade.calculate_interest(interest_rate=0.00025)) == 7.5414886436454e-06
+
+
+@pytest.mark.usefixtures("init_persistence")
+def test_interest_binance_lev(market_lev_buy_order, fee):
+    """
+        Market trade on Kraken at 3x and 5x leverage
+        Short trade
+        interest_rate: 0.05%, 0.25% per 4 hrs
+        open_rate: 0.00001099 base
+        close_rate: 0.00001173 base
+        stake_amount: 0.0009999999999226999
+        borrowed: 0.0019999999998453998
+        amount:
+            90.99181073 * leverage(3) = 272.97543219 crypto
+            90.99181073 * leverage(5) = 454.95905365 crypto
+        borrowed:
+            0.0019999999998453998  base
+            0.0039999999996907995  base
+        time-periods: 10 minutes(rounds up to 1/24 time-period of 24hrs)
+                        5 hours = 5/24
+
+        interest: borrowed * interest_rate * time-periods
+                = 0.0019999999998453998 * 0.00050 * 1/24 = 4.166666666344583e-08  base
+                = 0.0019999999998453998 * 0.00025 * 5/24 = 1.0416666665861459e-07 base
+                = 0.0039999999996907995 * 0.00050 * 5/24 = 4.1666666663445834e-07 base
+                = 0.0039999999996907995 * 0.00025 * 1/24 = 4.166666666344583e-08  base
+    """
+
+    trade = Trade(
+        pair='ETH/BTC',
+        stake_amount=0.0009999999999226999,
+        amount=272.97543219,
+        open_rate=0.00001099,
+        open_date=datetime.utcnow() - timedelta(hours=0, minutes=10),
+        fee_open=fee.return_value,
+        fee_close=fee.return_value,
+        exchange='binance',
+        leverage=3.0,
+        interest_rate=0.0005,
+        interest_mode=InterestMode.HOURSPERDAY
+    )
+    # 10 minutes round up to 4 hours evenly on kraken so we can predict the them more accurately
```

Comment:
Is this comment accurate in this case (this is the binance test)

Reply:
That comment is outdated